File: nodejs/src/state/ReadyState.ts
import OrderState, { IOrder } from "./OrderState";
import DeliveredState from "./DeliveredState";

class ReadyState extends OrderState {
    constructor(order: IOrder) { super(order); }

    getName(): string { return "READY"; }

    next(): void {
        console.log(`🚀 Đơn #${this.order.id}: READY → DELIVERED`);
        this.order.setState(new DeliveredState(this.order));
    }

    cancel(): void {
        console.log(`⚠️ Đơn #${this.order.id}: Không thể huỷ khi đã READY`);
    }
}

export default ReadyState;

File: spring/demo/src/main/java/com/example/demo/state/ReadyState.java
package com.example.demo.state;

import com.example.demo.model.Order;

public class ReadyState extends OrderState {
    public ReadyState(Order order) {
        super(order);
    }

    @Override
    public String getName() {
        return "READY";
    }

    @Override
    public void next() {
        System.out.println("Đơn #" + order.getId() + ": READY -> DELIVERED");
        order.setState(new DeliveredState(order));
    }

    @Override
    public void cancel() {
        System.out.println("Đơn #" + order.getId() + ": Không thể hủy khi đã READY");
    }
}
